numpy/lib/type_check.py

```diff
@@ -266,7 +272,19 @@ def iscomplexobj(x):
     True
 
     """
-    return issubclass(asarray(x).dtype.type, _nx.complexfloating)
+    try:
+        dtype = x.dtype
```

Comment:
If an exception does not get raised here, I'm pretty sure nothing gets returned from this function.

This is good example of why we insist on unit tests :)